govcd/vm_dhcp_test.go

```diff
@@ -0,0 +1,227 @@
+// +build nsxv vm functional ALL
+
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+// Test_VMGetDhcpAddress proves that it is possible to wait until DHCP lease is acquired by VM and
+// report the IP even if VM does not have guest tools installed.
+// The test does below actions:
+// 1. Creates VM
+// 2. Ensures there is a DHCP configuration for network
+// 3. Backs up network configuration for VM
+// 4. Powers off VM
+// 5. Sets VM network adapter to use DHCP
+// 6. Powers on VM and checks for a DHCP lease assigned to VM
+// 7. Cleans up
+func (vcd *TestVCD) Test_VMGetDhcpAddress(check *C) {
+	if vcd.config.VCD.EdgeGateway == "" {
+		check.Skip("Skipping test because no edge gateway given")
+	}
+
+	// Construct new VM for test
+	vapp, err := vcd.createTestVapp("GetDhcpAddress")
```

Comment:
It is defined inside the function `createTestVapp` and also at the bottom of this test function (just to avoid overloading test environment during multiple tests)